File: java/sci-java/src/com/company/I_sci_2.java
package com.company;

public class I_sci_2 {

    public static void main(String[] args) {
        int ITERATIONS = 100000000;
        int seq;
        double pi;
        long beginTime, endTime, timePassed;
        for (seq = 1; seq <= 5; seq++) {
            beginTime = System.currentTimeMillis();
            pi = calcPi(ITERATIONS);
            endTime = System.currentTimeMillis();
            timePassed = endTime - beginTime;
            System.out.printf("pi = %.10f calculated in %dms\n", pi, timePassed);
        }


    }

    private static double calcPi(int iterations) {
        double result = 0;
        int n;
        for (n = 0; n < iterations; n++) {
            result += Math.pow(-1, n) / (2 * n + 1);
        }
        return result * 4;
    }
}

File: js/sci-js/I.sci-2.js
let timeCosts = [];
const ITERATIONS = 100000000;

function calcPi() {
  let result = 0;
  for (let n = 0; n < ITERATIONS; n++) {
    result += Math.pow(-1, n) / (2 * n + 1);
  }
  return result * 4
}

for (let seq = 1; seq <= 5; seq++) {
  let timeBegin = new Date().getTime();
  let pi =calcPi();
  let timeEnd = new Date().getTime();
  let timePassed = timeEnd-timeBegin;
  timeCosts.push(timePassed);
  console.log(`pi = ${pi} calculated in ${timePassed}ms`)
}
